linked_list/implementation.py

```diff
@@ -1,40 +1,127 @@
 from .interface import AbstractLinkedList
-
+from .node import Node
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
 
     def __init__(self, elements=None):
-        pass
+        self.start = None
+        self.end = None
+        if elements:
+            for e in elements:
+                node = Node(e)
+                if not self.start:
+                    self.start = node
+                if self.end:
+                    self.end.next = node
+                self.end = node
 
     def __str__(self):
-        pass
+        str_list = []
+        for i in iter(self):
+            str_list.append(i.elem)
+        return str(str_list)
 
     def __len__(self):
-        pass
+        total = 0
+        for item in iter(self):
+            total += 1
+        return total
 
     def __iter__(self):
-        pass
+        if self.start:
+            element = self.start
+            while element.next:
+                yield element
+                element = element.next
+            # yield end element
+            yield element
+        
 
     def __getitem__(self, index):
-        pass
+        for i, element in enumerate(iter(self)):
+            if i == index:
+                return element
+        raise IndexError
 
     def __add__(self, other):
-        pass
+        new_list = LinkedList()
```

Comment:
OH! 😄 Then you really should have done the same thing on line 22 in your `__str__` function.